### PR TITLE
feat: added the functionality to save images from the clipboard to PN…

### PR DESCRIPTION
**Pull request by huaiyinfeilong**
* Added the functionality to save images from the clipboard to PNG files using the hotkey NVDA+Alt+Print Screen.
* Modified the clipboard editor to save images in PNG format.

### CODE REVIEW

File: addon/globalPlugins/clipboardEnhancement/__init__.py [hunk 1]
```diff
@@ -893,3 +897,37 @@ def script_emptyClipboardHistory(self, gesture):
...
+
+	@scriptHandler.script(
+		# Translators: Save the image in clipboard to file
+		description=_("将剪贴板图片另存为文件"),
+		gestures=["kb(desktop):NVDA+alt+printscreen", "kb(laptop):NVDA+alt+printscreen"])
+	def script_saveClipboardImageToFile(self, gesture):
+		def saveClipboardImageToFile():
+			import sys
+			import os
+			sys.path.append(
```
Comment (cary-rowen) [inline addon/globalPlugins/clipboardEnhancement/__init__.py:909]: Would you mind moving the import to the top?

File: addon/globalPlugins/clipboardEnhancement/__init__.py [hunk 2]
```diff
@@ -893,3 +897,37 @@ def script_emptyClipboardHistory(self, gesture):
...
+				os.path.abspath(
+				os.path.join(os.path.dirname(__file__), "..", "_py3_contrib")
+				)
+			)
+			from PIL import Image, ImageGrab
+			from io import BytesIO
+			try:
```
Comment (cary-rowen) [inline addon/globalPlugins/clipboardEnhancement/__init__.py:916]: Ditto, I think this could also be moved to the top

**Comment by huaiyinfeilong** [comment]
"I think it can be moved to the top. Previously, there was an issue because we weren't using the VSCode configuration within the plugin template, which caused a warning to appear when moved to the top, specifically after os.sys…"

 

发件人: ***@***.*** ***@***.***> 代表 Rowen
发送时间: 2024年5月24日 20:07
收件人: cary-rowen/clipboardEnhancement ***@***.***>
抄送: huaiyinfeilong ***@***.***>; Author ***@***.***>
主题: Re: [cary-rowen/clipboardEnhancement] feat: added the functionality to save images from the clipboard to PN… (PR #16)

 

@cary-rowen commented on this pull request.

  _____  

In addon/globalPlugins/clipboardEnhancement/__init__.py <https://github.com/cary-rowen/clipboardEnhancement/pull/16#discussion_r1613375964> :

> +                    import sys
+                      import os
+                      sys.path.append(

Would you mind moving the import to the top?

**Comment by huaiyinfeilong** [comment]
"I think it can be moved to the top. Previously, there was an issue because we weren't using the VSCode configuration within the plugin template, which caused a warning to appear when moved to the top, specifically after os.sys…"

 

发件人: ***@***.*** ***@***.***> 代表 Rowen
发送时间: 2024年5月24日 20:09
收件人: cary-rowen/clipboardEnhancement ***@***.***>
抄送: huaiyinfeilong ***@***.***>; Author ***@***.***>
主题: Re: [cary-rowen/clipboardEnhancement] feat: added the functionality to save images from the clipboard to PN… (PR #16)

 

@cary-rowen commented on this pull request.

  _____  

In addon/globalPlugins/clipboardEnhancement/__init__.py <https://github.com/cary-rowen/clipboardEnhancement/pull/16#discussion_r1613378473> :

> +                    from PIL import Image, ImageGrab
+                      from io import BytesIO
+                      try:

Ditto, I think this could also be moved to the top

**Comment by huaiyinfeilong** [comment]
"I think it can be moved to the top. Previously, there was an issue because we weren't using the VSCode configuration within the plugin template, which caused a warning to appear when moved to the top, specifically after os.sys…"

 

发件人: ***@***.*** ***@***.***> 代表 Rowen
发送时间: 2024年5月24日 20:13
收件人: cary-rowen/clipboardEnhancement ***@***.***>
抄送: huaiyinfeilong ***@***.***>; Author ***@***.***>
主题: Re: [cary-rowen/clipboardEnhancement] feat: added the functionality to save images from the clipboard to PN… (PR #16)

 

@cary-rowen commented on this pull request.

  _____  

In addon/globalPlugins/clipboardEnhancement/clipEditor.py <https://github.com/cary-rowen/clipboardEnhancement/pull/16#discussion_r1613383292> :

> +            import sys
+              import os
+              sys.path.append(
+                      os.path.abspath(
+                      os.path.join(os.path.dirname(__file__), "..", "_py3_contrib")
+                      )
               )
+              from PIL import Image, ImageGrab
+              from io import BytesIO
               try:

This also does the same thing, i.e. moves the imports to the top

**Comment by cary-rowen** [comment]
Thanks @huaiyinfeilong 
I will adjust the import separately.

**Pull request merged**